src/seas5/mars_tprate.py

```diff
@@ -0,0 +1,147 @@
+            for coord in [
+                bbox[3],
+                bbox[0],
+                bbox[1],
+                bbox[2],
+            ]
+        ]
+    )
+
+    path_raw = os.path.join(dir, f"seas5_mars_tprate_{year}.grib")
+    temp_base = os.path.basename(path_raw)
+    raw_outpath = os.path.join("mars", "raw", temp_base)
+
+    # Generate a sequence of monthly dates
+    start_date = pd.to_datetime(f"{year}-01-01")
+    end_date = pd.to_datetime(f"{year}-12-01")
+    date_range = pd.date_range(start=start_date, end=end_date, freq="MS")
+    date_strings = [date.strftime("%Y-%m-%d") for date in date_range]
+    dates_use = "/".join(date_strings)
+
+    # Pre 2016 has 25 ensemble members and then 51 afterwards
+    if year <= 2016:
+        ensemble_members = "/".join([str(i) for i in range(25)])
+    else:
+        ensemble_members = "/".join([str(i) for i in range(51)])
+
+    # See docs for more details on parameters: 
+    # https://confluence.ecmwf.int/display/UDOC/Keywords+in+MARS+and+Dissemination+requests?src=contextnavpagetreemode
+    server.execute(
+        {
+            "class": "od", # operational archive
+            "date": dates_use,
+            "expver": "0001", # model version
+            "fcmonth": "1/2/3/4/5/6/7", # forecast months
+            "levtype": "sfc", # surface horizontal level
+            "method": "1",
+            "area": bbox_str,
+            "grid": "0.4/0.4",
+            "number": ensemble_members,
+            "origin": "ecmf",
+            "param": "228.172", # tprate
+            "stream": "msmm", # multi-model seasonal forecast atmospheric monthly means
+            "system": "5",
+            "time": "00:00:00",
+            "type": "fcmean", # forecast mean
+            "target": "output",
+        },
+        path_raw,
+    )
+    print("... Data downloaded successfully. Uploading data to Azure...")
+    if save_to_cloud:
+        upload_file(
+            local_file_path=path_raw,
+            sas_token=SAS_TOKEN_DEV,
+            container_name=CONTAINER_GLOBAL,
+            storage_account=STORAGE_ACCOUNT_DEV,
+            blob_path=raw_outpath,
+        )
+        print(
+            f"... Data uploaded successfully to Azure. Saved temporarily to {path_raw}"
+        )
+
+    return path_raw
+
+
+def process_archive(path_raw, dir, save_to_cloud=True):
+    """
+    Processes raw grib files to output analysis-ready COGs (.tif)
+
+    Args:
+        path_raw (str): Location of the input raw data
+        dir (str): (Temporary) Location to save the data
+        save_to_cloud (Bool): Whether to save the processed .tif files to cloud storage
+    
+    Returns:
+        None
+    """
+
+    print(f"Processing temporary file: {path_raw}...")
+    ds = xr.open_dataset(
+        path_raw,
+        engine="cfgrib",
+        drop_variables=["surface", "values"],
+        backend_kwargs=dict(time_dims=("time", "forecastMonth")),
+    )
+
+    ds_mean = ds.mean(dim="number")
```

Comment:
It seems like after taking the mean, we lose all the attributes of the `tprate` variable. We don't need most of them, but it would good to keep at least the `long_name` (Mean total precipitation rate) and `units` (m s**-1):

<img width="326" alt="image" src="https://github.com/user-attachments/assets/2aa0816b-fb02-4714-84ab-73e073fcf14e">

We can copy all the attributes over with `ds_mean["tprate"].attrs = ds["tprate"].attrs`, or just do for a subset if we don't want all the 32 existing attributes. But then at least we can immediately see the attributes when we load the COG.

Reply:
Good catch! Just updated. Seems not a bad idea to just include all of them!